app/controllers/provider/sessions_controller.rb

```diff
@@ -33,6 +33,11 @@ def create
     end
   end
 
+  def bounce
+    auth = AuthenticationProvider.where(account_type: 'provider').find(params[:authorization_provider_id])
+    redirect_to OauthFlowPresenter.new(auth, request).authorize_url
```

Comment:
I am not sure if this should be `OauthFlowPresenter` or `ProviderOauthFlowPresenter`. The difference would be the generated `redirect_uri`

Reply:
This should be ProviderOauthFlowPresenter as it is Admin SSO

Reply:
Probably using the system name instead of the id as well